### PR TITLE
Bug 2086519: AUTH-133: test/extended/security/scc: comply to pod security admission

### PR DESCRIPTION
**Pull request by s-urbaniak**
Currently, namespaces created via `CLI#CreateProject()` don't respect the configured pod security profile. This fixes it.

Further, this lets pass i.e. the e2e test: `[sig-auth][Feature:SecurityContextConstraints]  TestAllowedSCCViaRBAC [Suite:openshift/conformance/parallel]` as part of https://github.com/openshift/cluster-kube-apiserver-operator/pull/1339.

/cc @stlaz

### CODE REVIEW

File: test/extended/util/client.go [hunk 1]
```diff
@@ -434,9 +412,39 @@ func (c *CLI) CreateProject() string {
...
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// once permissions are settled the underlying namespace must have been created.
+		ns, err := c.AdminKubeClient().CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if len(c.kubeFramework.NamespacePodSecurityEnforceLevel) == 0 {
+			// TODO(sur): set to restricted in a separate PR and fix failing tests
+			c.kubeFramework.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
```
[the remark this answers is not in the document]
Comment (s-urbaniak) [reply test/extended/util/client.go]: done, let the hunt start

**Comment by s-urbaniak** [comment]
so far e2e-aws-csi and aws-single-node failures seem unrelated 🤔

**Comment by andreaskaris** [comment]
But the serial test shows a few issues:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27154/pull-ci-openshift-origin-master-e2e-aws-serial/1529063550553690112

~~~
:  [sig-network] services when running openshift ipv4 cluster ensures  external ip policy is configured correctly on the cluster [Serial]  [Suite:openshift/conformance/serial] expand_less | 4s
-- | --
{  fail [github.com/openshift/origin/test/extended/networking/util.go:607]: failed to create new pod in namespace: e2e-test-ns-global-grrcg Unexpected error:     <*errors.StatusError \| 0xc002a31c20>: {         ErrStatus: {             TypeMeta: {Kind: "", APIVersion: ""},             ListMeta: {                 SelfLink: "",                 ResourceVersion: "",                 Continue: "",                 RemainingItemCount: nil,             },             Status: "Failure",             Message: "pods \"test-ip-family-podh2rj9\" is forbidden: violates PodSecurity \"restricted:latest\": allowPrivilegeEscalation != false (container \"agnhost-container\" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container \"agnhost-container\" must set securityContext.capabilities.drop=[\"ALL\"]), runAsNonRoot != true (pod or container \"agnhost-container\" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container \"agnhost-container\" must set securityContext.seccompProfile.type to \"RuntimeDefault\" or \"Localhost\")",             Reason: "Forbidden",             Details: {                 Name: "test-ip-family-podh2rj9",                 Group: "",                 Kind: "pods",                 UID: "",                 Causes: nil,                 RetryAfterSeconds: 0,             },             Code: 403,         },     }     pods "test-ip-family-podh2rj9" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "agnhost-container" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "agnhost-container" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "agnhost-container" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "agnhost-container" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost") occurred}
~~~

The same for the 3 EgressIP tests that were run:
~~~
:  [sig-network][Feature:EgressIP] [external-targets] pods should have the  assigned EgressIPs and EgressIPs can be deleted and recreated  [Skipped:azure] [Serial] [Suite:openshift/conformance/serial] expand_more               13s | :  [sig-network][Feature:EgressIP] [external-targets] pods should have the  assigned EgressIPs and EgressIPs can be deleted and recreated  [Skipped:azure] [Serial] [Suite:openshift/conformance/serial] expand_more | 13s
:  [sig-network][Feature:EgressIP] [external-targets] pods should keep the  assigned EgressIPs when being rescheduled to another node [Serial]  [Suite:openshift/conformance/serial] expand_more | 13s
:  [sig-network][Feature:EgressIP] [external-targets] EgressIPs can be  assigned automatically [Skipped:Network/OVNKubernetes] [Serial]  [Suite:openshift/conformance/serial] expand_more | :  [sig-network][Feature:EgressIP] [external-targets] EgressIPs can be  assigned automatically [Skipped:Network/OVNKubernetes] [Serial]  [Suite:openshift/conformance/serial] expand_more
~~~

Can you fix this, or what's needed to fix this?

**Comment by s-urbaniak** [comment]
@andreaskaris the egress e2e tests needed privileges adjusted.

**Comment by s-urbaniak** [comment]
https://github.com/openshift/cluster-kube-apiserver-operator/pull/1358 has merged. Resetting all previously labelled `privileged` e2e tests back to `baseline` to iterate on the fine tuning.

**Comment by s-urbaniak** [comment]
@stlaz @deads2k after looking at the logs I don't see the failures are being related to the change here, ptal.

**Comment by stlaz** [comment]
I meant "superseded by #27269"

**Pull request closed without merging**